### PR TITLE
Incremental strict null checks adoption: Slice 12

### PR DESCRIPTION
**Pull request by fregante**
Very straightforward this time

## What does this PR do?

- Follows https://github.com/pixiebrix/pixiebrix-extension/pull/6847

## Checklist

- [x] Designate a primary reviewer: @grahamlangford

### CODE REVIEW

File: src/components/AsyncStateGate.tsx [hunk 1]
```diff
@@ -97,7 +98,8 @@ const AsyncStateGate = <Data,>(
     throw error;
   }
 
+  assert(data, "data should be defined"); // Type-only check
   return <>{children({ data })}</>;
```
Comment (fregante) [inline src/components/AsyncStateGate.tsx:102]: From what I understand, `data` is only undefined while it's loading or it's an error, so it's already covered by the previous condition.

File: src/components/fields/schemaFields/widgets/cssClassWidgets/utils.ts [hunk 2]
```diff
@@ -71,12 +70,10 @@ export function parseValue(value: Value): {
...
-  reportError(
-    new Error(
-      `Unexpected value parsing the CSS class. Type of value: ${typeof value}. Value: ${JSON.stringify(
-        value
-      )}`
-    )
+  throw new Error(
+    `Unexpected value parsing the CSS class. Type of value: ${typeof value}. Value: ${JSON.stringify(
+      value
+    )}`
```
Comment (fregante) [inline src/components/fields/schemaFields/widgets/cssClassWidgets/utils.ts:76]: We should refrain from using `reportError` directly, let errors bubble up instead.

In this case, it seems that all the code expected the return value to _not_ be `undefined`, so by not throwing here you'd get an additional error downstream.

File: src/extensionConsole/pages/mods/gridView/GridCardErrorBoundary.tsx [hunk 3]
```diff
@@ -37,15 +37,16 @@ type Props = {
...
 
 class GridCardErrorBoundary extends Component<Props, State> {
-  constructor(props: Props) {
-    super(props);
-    this.state = { hasError: false, errorMessage: undefined, stack: undefined };
-  }
+  override state: State = {
+    hasError: false,
+    errorMessage: undefined,
+    stack: undefined,
```
Comment (fregante) [inline src/extensionConsole/pages/mods/gridView/GridCardErrorBoundary.tsx:48]: I saw several copy-pasted abcErrorBoundary components, should I extract an `ExtendableErrorBoundary` instead?

File: src/hooks/useAsyncExternalStore.ts [hunk 4]
```diff
@@ -24,47 +24,43 @@ import {
...
 
 class StateController<T = unknown> {
   private readonly stateListeners = new Set<() => void>();
   private state: AsyncState<T> = uninitializedAsyncStateFactory();
-  private nonce: UUID;
-
-  // Methods to pass to useSyncExternalStore
-  readonly boundSubscribe: Subscribe;
-  readonly boundGetSnapshot: () => AsyncState<T>;
+  private nonce: UUID = validateUUID(null); // TODO: Drop after strictNullCheck transition; Silences TS bug
```
Comment (fregante) [inline src/hooks/useAsyncExternalStore.ts:35]: TypeScript was generating an error, but just on the strict config. The error was a bug/limitation of TS, so I'm just silencing it for now and hoping for the best in the future.


> Property 'nonce' has no initializer and is not definitely assigned in the constructor

File: src/components/AsyncStateGate.tsx (hunk 1 above)
Comment (grahamlangford) [reply src/components/AsyncStateGate.tsx:102]: Theoretically we could AsyncStateGate an API request that could return a 200 without a payload. I don't know of any instances where that could happen now, but this wouldn't be future-proof in that case.

File: src/extensionConsole/pages/mods/gridView/GridCardErrorBoundary.tsx (hunk 3 above)
Comment (grahamlangford) [reply src/extensionConsole/pages/mods/gridView/GridCardErrorBoundary.tsx:48]: I suggest discussing with @BLoe. He's going to be looking into improving our error telemetry, and part of that will likely involve revisiting our ErrorBoundaries.

File: src/components/AsyncStateGate.tsx (hunk 1 above)
Comment (BLoe) [reply src/components/AsyncStateGate.tsx:102]: I think in that case we'd want to also tweak the types to more explicitly allow `void` data in the AsyncStateGate anyway, so I'm fine with this assertion for now.

**Review by grahamlangford — APPROVED** [review]

**Pull request merged**